### PR TITLE
Updates for TLS: reading from encrypted PEM file & hostname fix

### PR DESCRIPTION
**Pull request by aegal**
This change is needed to support recent TLS changes
1) Read from Encrypted PEM file
2) Read hostname from CNS commands if provided

### CODE REVIEW

File: server/tls/tlscertificate_retriever_linux.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+// +build linux
```
Comment (ramiro-gamarra) [inline server/tls/tlscertificate_retriever_linux.go:1]: If the file is suffixed with `linux`, then the build tag is not needed

File: server/tls/tlscertificate_retriever_linux_test.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// +build linux
```
Comment (ramiro-gamarra) [inline server/tls/tlscertificate_retriever_linux_test.go:1]: same as above

File: server/tls/tlscertificate_retriever_windows_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// +build windows
```
Comment (ramiro-gamarra) [inline server/tls/tlscertificate_retriever_windows_test.go:1]: same as above. relevant read: https://golang.org/cmd/go/#hdr-Build_constraints

File: server/tls/tlscertificate_retriever_windows.go [hunk 4]
```diff
@@ -1,77 +1,140 @@
+// +build windows
```
Comment (ramiro-gamarra) [inline server/tls/tlscertificate_retriever_windows.go:1]: same as above

File: server/tls/tls_certificate_retriever.go [hunk 5]
```diff
@@ -8,8 +8,11 @@ import (
...
 
 // TlsCertificateRetriever is the interface used by
-// both windows cert store implementation, linux and cert from file retriever.
+// both windows and linux and cert from file retriever.
 type TlsCertificateRetriever interface {
 	GetCertificate() (*x509.Certificate, error)
 	GetPrivateKey() (crypto.PrivateKey, error)
+	ReadFile() ([]byte, error)
+	Decrypt([]byte)  (string, error)
+	ParsePEMFile([]byte) (error)
```
Comment (ramiro-gamarra) [inline server/tls/tls_certificate_retriever.go:17]: It seems the new interface methods are unused, and can remain internal to the implementations.

File: server/tls/tlscertificate_retriever_linux.go [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+		return fmt.Errorf("Invalid PEM format located at %s", fcert.settings.TLSCertificatePath)
+	}
+
+	fcert.pemBlock = pemBlocks
+	return nil
+}
+
+// ReadFile reads a from disk
+// linux uses file permissions and does not need to encrypt the file
+func (fcert *linuxTlsCertificateRetriever) Decrypt(content []byte) (string, error) {
```
Comment (ramiro-gamarra) [inline server/tls/tlscertificate_retriever_linux.go:90]: Once the interface is shortened, this method won't be needed

**Review by ramiro-gamarra — COMMENTED** [review]
A few comments

File: server/tls/tlscertificate_retriever_linux.go (hunk 1 above)
Comment (aegal) [reply server/tls/tlscertificate_retriever_linux.go]: Thanks, I removed it and I added a comment to this to make it easier for future readers of these files.

File: server/tls/tls_certificate_retriever.go (hunk 5 above)
Comment (aegal) [reply server/tls/tls_certificate_retriever.go]: Yeah, good point, I removed it from the interface.

File: server/tls/tlscertificate_retriever_linux.go (hunk 6 above)
Comment (aegal) [reply server/tls/tlscertificate_retriever_linux.go]: Removed

**Review by ramiro-gamarra — APPROVED** [review]
:shipit:

**Pull request merged**